components/renku_data_services/message_queue/redis_queue.py

```diff
@@ -115,65 +100,54 @@ def decorator(f):
+                case v1.ProjectRemoved | v2.ProjectRemoved:
                     queue_name = "project.removed"
-                case UserAdded.__qualname__:
+                case v1.UserAdded | v2.UserUpdated:
                     queue_name = "user.added"
-                case UserUpdated.__qualname__:
+                case v1.UserUpdated | v2.UserUpdated:
                     queue_name = "user.updated"
-                case UserRemoved.__qualname__:
+                case v1.UserRemoved | v2.UserRemoved:
                     queue_name = "user.removed"
-                case ProjectAuthorizationAdded.__qualname__:
-                    queue_name = "projectAuth.added"
-                case ProjectAuthorizationUpdated.__qualname__:
-                    queue_name = "projectAuth.updated"
-                case ProjectAuthorizationRemoved.__qualname__:
+                case AmbiguousEvent.PROJECT_MEMBERSHIP_CHANGED:
+                    queue_name = "to_be_determined_later"
+                case v2.ProjectMemberRemoved:
                     queue_name = "projectAuth.removed"
                 case _:
-                    raise NotImplementedError(f"Can't create message using transform {transform}:{signature}")
-            headers = create_header(queue_name)
-            message_id = ULID().hex
-            message: dict[bytes | memoryview | str | int | float, bytes | memoryview | str | int | float] = {
-                "id": message_id,
-                "headers": headers.serialize_json(),
-                "payload": base64.b64encode(serialize_binary(payload)).decode(),
-            }
-            event_id = await self.event_repo.store_event(session, queue_name, message)
-            await session.commit()
-
-            try:
-                await self.message_queue.send_message(queue_name, message)
-            except:  # noqa:E722
-                return result
-            await self.event_repo.delete_event(event_id)
+                    raise NotImplementedError(f"Can't find queue name for event type {event_type}")
+            for payload in payloads:
+                if isinstance(payload, v2.ProjectMemberUpdated):
+                    queue_name = "projectAuth.updated"
+                elif isinstance(payload, v2.ProjectMemberAdded):
+                    queue_name = "projectAuth.added"
+                elif isinstance(payload, v2.ProjectMemberRemoved):
+                    queue_name = "projectAuth.removed"
+                message_id = ULID().hex
+                headers = create_header(queue_name).serialize_json()
+                message: dict[bytes | memoryview | str | int | float, bytes | memoryview | str | int | float] = {
+                    "id": message_id,
+                    "headers": headers,
+                    "payload": base64.b64encode(serialize_binary(payload)).decode(),
+                }
+                event_id = await self.event_repo.store_event(session, queue_name, message)
```

Comment:
If you commit here then I cannot undo the transaction in the authorization decorator if something fails. So that should be fixed in cooldown. And for now it is more important for the authz service to be consistent in this case than for the search (via the message queue).

Reply:
Opened an issue #189 to clean this up properly.